src/dump.c

```diff
@@ -678,7 +680,7 @@ static void jl_serialize_code_instance(jl_serializer_state *s, jl_code_instance_
     if (write_ret_type && codeinst->rettype_const &&
             jl_typeis(codeinst->rettype_const, jl_partial_opaque_type)) {
         if (skip_partial_opaque) {
-            jl_serialize_code_instance(s, codeinst->next, skip_partial_opaque, internal);
+            jl_serialize_code_instance(s, codeinst->next, skip_partial_opaque, internal, 0);
```

Comment:
Why not pass `force` here?

Reply:
I'm just preserving how this function worked before. Presumably there was a reason that `jl_serialize_code_instance` didn't serialize CIs with backref entries on recursive calls to `jl_serialize_code_instance`, so I'm just preserving that behavior. Instead, I'm only forcing the CIs to be serialized when I know that the backref table initialization doesn't mean that the object has been serialized, i.e., from the `jl_is_code_instance` branch in `jl_serialize_value`.